### PR TITLE
Add necessary Documenter.jl files

### PR DESCRIPTION
**Pull request by GordStephen**
Adds in the files necessary for implementing Documenter.jl. @mbauman will still need to do two steps ([#1](https://juliadocs.github.io/Documenter.jl/latest/man/hosting/#deploy-keys), [#2](https://juliadocs.github.io/Documenter.jl/latest/man/hosting/#travis-environment-settings)) before it's up and running.

For now I just set the doc site homepage to have the same content as the README and created a second auto-generated reference page to display the package's docstrings - this can no doubt be improved on moving forward.

### CODE REVIEW

**Comment by mbauman** [comment]
Sorry I'm slow on this! Is it possible for me to give you the necessary permissions here?

**Comment by GordStephen** [comment]
No rush on my end - the blocking step would be adding a deploy key (for Travis) to the repository, which can only be done by the repository owner AFAIK.

File: .travis.yml [hunk 1]
```diff
@@ -13,4 +13,6 @@ script:
   - julia -e 'Pkg.clone("https://github.com/mbauman/Tuples.jl"); Pkg.clone("https://github.com/mbauman/RangeArrays.jl")'
   - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("AxisArrays"); Pkg.test("AxisArrays"; coverage=true)'
 after_success:
-    - julia -e 'cd(Pkg.dir("AxisArrays")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
+  - julia -e 'cd(Pkg.dir("AxisArrays")); Pkg.add("Coverage"); using Coverage; Coveralls.submit(Coveralls.process_folder())'
+  - julia -e 'Pkg.add("Documenter")'
+  - julia -e 'cd(Pkg.dir("AxisArrays")); include(joinpath("doc", "make.jl"))'
```
Comment (GordStephen) [inline .travis.yml]: "doc" -> "docs" will need to be changed here too

Comment (mbauman) [inline .travis.yml]: 👍  thanks!  I think I got all the keys set up now, too, so we'll see if it works after this push.  If you'd prefer, I can now move the folder back, but the Documenter setup tasks seemed to assume this layout… so this was the path of least resistance.

Comment (GordStephen) [inline .travis.yml]: Changing it makes sense, best to stick with the Documenter defaults unless there's a reason to deviate I think.

**Comment by mbauman** [comment]
Hm, I don't think that last step is necessary (it errors and other packages don't seem to do it).  I think this just needs to be on master for it to work now — Documenter.jl simply reports "skipping deployment".

Ready to merge?

**Comment by GordStephen** [comment]
For sure! :+1:

**Comment by GordStephen** [comment]
Lots of room for improvement, but it's a start: http://mbauman.github.io/AxisArrays.jl/latest/

**Pull request merged**